pkg/reconciler/broker/broker.go

```diff
@@ -428,6 +435,55 @@ func (r *Reconciler) reconcileChannel(ctx context.Context, channelResourceInterf
 	return channelable, nil
 }
 
+func (r *Reconciler) reconcileBrokerChannelEventPolicies(ctx context.Context, b *eventingv1.Broker, triggerChan *duckv1.Channelable, featureFlags feature.Flags) error {
+	expected := resources.MakeEventPolicyForBackingChannel(b, triggerChan)
+
+	if featureFlags.IsOIDCAuthentication() {
+		// get the eventpolicy , create if not exists.
+		foundEP, err := r.eventPolicyLister.EventPolicies(expected.Namespace).Get(expected.Name)
+		if apierrs.IsNotFound(err) {
+			// create the EventPolicy if it doesn't exists.
+			logging.FromContext(ctx).Info("Creating EventPolicy for Broker %s", expected.Name)
+
+			_, err = r.eventingClientSet.EventingV1alpha1().EventPolicies(expected.Namespace).Create(ctx, expected, metav1.CreateOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to create EventPolicy for Broker %s: %w", expected.Name, err)
+			}
+		} else if err != nil {
+			return fmt.Errorf("failed to get EventPolicy for Broker %s: %w", expected.Name, err)
+		} else if r.policyNeedsUpdate(foundEP, expected) {
+			// update the EventPolicy if it exists and needs update.
+			logging.FromContext(ctx).Info("Updating EventPolicy for Broker %s", expected.Name)
+			expected.SetResourceVersion(foundEP.GetResourceVersion())
+			_, err = r.eventingClientSet.EventingV1alpha1().EventPolicies(expected.Namespace).Update(ctx, expected, metav1.UpdateOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to update EventPolicy for Broker %s: %w", expected.Name, err)
+			}
+		}
+	} else {
+		// list all the orphaned eventpolicies that have owner reference set to the broker and delete them.
+		eventPolicies, err := r.eventPolicyLister.EventPolicies(expected.Namespace).List(labels.Everything())
+		if err != nil {
+			return fmt.Errorf("failed to list EventPolicies for Broker %s: %w", expected.Name, err)
+		}
+		for _, ep := range eventPolicies {
+			if metav1.IsControlledBy(ep, b) {
+				logging.FromContext(ctx).Info("Deleting EventPolicy for Broker %s", expected.Name)
+				err := r.eventingClientSet.EventingV1alpha1().EventPolicies(ep.Namespace).Delete(ctx, ep.Name, metav1.DeleteOptions{})
+				if err != nil {
+					return fmt.Errorf("failed to delete EventPolicy for Broker %s: %w", expected.Name, err)
+				}
+				logging.FromContext(ctx).Info("Deleted EventPolicy for Broker %s", expected.Name)
+			}
+		}
+	}
+	return nil
+}
+
+func (r *Reconciler) policyNeedsUpdate(foundEP, expected *eventingv1alpha1.EventPolicy) bool {
```

Comment:
nit: this function doesn't really need to be a method on the Reconciler struct
```suggestion
func policyNeedsUpdate(foundEP, expected *eventingv1alpha1.EventPolicy) bool {
```

Reply:
done, PTAL!